### PR TITLE
Fix rpm build

### LINKED ISSUE #22231: make rpm fails
Attempted on both OSX and Ubuntu 18.04, both with the same failure:
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"tools/packaging/rpm/build-istio-rpm.sh\": stat tools/packaging/rpm/build-istio-rpm.sh: no such file or directory": unknown.
tools/packaging/rpm/rpm.mk:4: recipe for target 'rpm/istio' failed
```

### PR DESCRIPTION
**Pull request by mkretz**
Fixes #22231

With these changes the rpm/istio and rpm/proxy targets work again. I still need to test that the resulting RPMs work correctly. Want to open this draft PR in order to get some more eyes on this, as my fix was rather explorative. Can someone do a preliminary review to verify that I am on the right track here?

### CODE REVIEW

**Comment by mkretz** [comment]
@googlebot I signed it!

File: tools/packaging/rpm/istio/istio.spec [hunk 1]
```diff
@@ -37,9 +37,7 @@ Source0:        istio.tar.gz
 #Source1:        istiorc
 #Source2:        buildinfo
 Source3:        istio-start.sh
-Source4:        istio-node-agent-start.sh
 Source5:        istio.service
-Source6:        istio-auth-node-agent.service
 
 # e.g. el6 has ppc64 arch without gcc-go, so EA tag is required
```
Comment (sdake) [inline tools/packaging/rpm/istio/istio.spec]: I know you didn't write this original packaging spec.

I answered some questions on stack overflow about building RPMS on RHEL 7.3 on PPC.

Does this build and work on ppc64 architectures?

See stack overflow question and answer here:

https://stackoverflow.com/questions/60394286/gcc-woverloaded-virtual-error-on-powerpc-ppc64le/60443777#60443777

and here:
https://stackoverflow.com/questions/60392262/bazel-tests-fail-on-powerpc-ppc64le-architecture/60695092#60695092

File: tools/packaging/rpm/istio/istio.spec [hunk 2]
```diff
@@ -37,9 +37,7 @@ Source0:        istio.tar.gz
...
 ExclusiveArch:  %{?go_arches:%{go_arches}}%{!?go_arches:%{ix86} x86_64 aarch64 %{arm}}
```
Comment (sdake) [inline tools/packaging/rpm/istio/istio.spec]: ExclusiveArch guidelines are here:
https://docs.fedoraproject.org/en-US/packaging-guidelines/#_arch_specific_runtime_and_build_time_dependencies

I think all this go_arches stuff seems nonsensical from a packaging point of view. Can you please sort out a proper ExclusiveArch line (or remove it entirely as per guidelines).

File: tools/packaging/rpm/istio/istio.spec [hunk 3]
```diff
@@ -231,14 +229,12 @@ install -d -m755 $RPM_BUILD_ROOT/%{_bindir}
 install -d -m755 $RPM_BUILD_ROOT/%{_unitdir}
 
 install -m755 %{SOURCE3} $RPM_BUILD_ROOT/%{_bindir}/istio-start.sh
-install -m755 %{SOURCE4} $RPM_BUILD_ROOT/%{_bindir}/istio-node-agent-start.sh
```
Comment (sdake) [inline tools/packaging/rpm/istio/istio.spec]: please renumber and clean up the source lines.

Comment (sdake) [reply tools/packaging/rpm/istio/istio.spec]: Please follow the fedora packaging guidelines related to %{buildroot} throughout this spec file: https://docs.fedoraproject.org/en-US/packaging-guidelines/#_using_buildroot_and_optflags_vs_rpm_build_root_and_rpm_opt_flags

File: tools/packaging/rpm/istio/istio.spec (hunk 2 above)
Comment (sdake) [reply tools/packaging/rpm/istio/istio.spec]: Please follow the fedora packaging guidelines for this spec file: https://docs.fedoraproject.org/en-US/packaging-guidelines/

File: tools/packaging/rpm/istio/istio.spec [hunk 4]
```diff
@@ -231,14 +229,12 @@ install -d -m755 $RPM_BUILD_ROOT/%{_bindir}
...
 
 install -m644 %{SOURCE5} $RPM_BUILD_ROOT/%{_unitdir}/istio.service
-install -m644 %{SOURCE6} $RPM_BUILD_ROOT/%{_unitdir}/istio-auth-node-agent.service
 
 binaries=(pilot-discovery pilot-agent istioctl sidecar-injector mixs mixc istio_ca galley node_agent istio-iptables istio-clean-iptables)
 pushd .
-cd ISTIO/out/linux_amd64/release
+cd ISTIO/src/istio.io/istio/out/linux_amd64
```
Comment (sdake) [inline tools/packaging/rpm/istio/istio.spec]: This conditional and shell script in RPM makes me sad. Can you take this PR a little further and ensure this RPM spec file could pass a fedora packaging review?

**Comment by mkretz** [comment]
@sdake thanks for your review. Much appreciated! To answer your initial point:

> I am not sure how far down the rabbit hole of RPM packaging you wish to proceed.

I am helping one of our clients in getting their application migrated to Istio and we need to extend the mesh to some RHEL VMs. That's why my number one priority is getting working RPMs in order to achieve this. Having the scripts comply with the Fedora guidelines sounds like a sound goal but I am definitely not a specialist in that area.

Can we first concentrate on getting the RPMs built again? Then I am happy to help with some further cleanup steps with your guidance. Makes sense?

I'll test the RPMs that I was able to produce with the proposed fixes and check whether the RHEL VM mesh expansion use case works using these RPMs.

**Comment by mkretz** [comment]
Tested the generated RPMs. Unfortunately, it seems that they don't install all of the necessary components. In particular they don't install the istio service which I am expecting to be able to launch on the vm using `service istio start`. Looks like we have to dive deeper into the workings of the build script.

**Comment by mkretz** [comment]
Out of curiosity I tried converting the `.deb` package which is shipped with the release to RPM using `alien` on RHEL 8. The conversion runs through, but it seems to yield an invalid RPM:
```
[root@envoy-test ~]# rpm -ivh istio-sidecar-1.5.0-2.x86_64.rpm
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
        file / from install of istio-sidecar-1.5.0-2.x86_64 conflicts with file from package filesystem-3.8-2.el8.x86_64
        file /lib from install of istio-sidecar-1.5.0-2.x86_64 conflicts with file from package filesystem-3.8-2.el8.x86_64
        installing package istio-sidecar-1.5.0-2.x86_64 needs 87MB on the /usr/local filesystem
```

**Comment by sdake** [comment]
Not sure what alien is although any conversion is doomed to fail. RPM is a very expressive packaging system and deb does not translate well to it.

**Pull request closed without merging**